wom/src/main/scala/wom/types/WomBooleanType.scala

```diff
@@ -10,6 +10,9 @@ case object WomBooleanType extends WomPrimitiveType {
 
   override protected def coercion = {
     case b: Boolean => WomBoolean(b)
+    case WomOptionalValue(WomBooleanType, Some(WomBoolean(b))) =>
+      println("Warning: you just coerced a filled Boolean optional to a non-optional Boolean and you should feel bad about yourself")
```

Comment:
🤔

Reply:
Similar in spirit to the cwltool error, minus the helpful context ("This happened with input 'foo'") and the use of a proper logging facility since there isn't one in scope here.

Reply:
If this PR goes beyond "DO NOT MERGE", I probably would tone down the judgy-ness though...

Reply:
gotcha, `LogLevel.JUDGY`.